-Implements sprint 06 according to requirements

tlavr opened the pull request:
-Updates questions generation using IMDB top250 films API
-Adds network handling

tlavr committed:
-Adds network handling

-Updates questions generation using IMDB top250 films API

-Implements sprint 06 according to requirements

NaR0RoW commented on MovieQuiz/Models/MostPopularMovies.swift:
```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+struct MostPopularMovies: Codable {
+    let errorMessage: String
```
Обращу внимание на свойство `errorMessage`.

Здесь может прийти текст с ошибкой, в таком случае сервис обычно возвращает пустой список фильмов `items`. С пустым списком приложение будет выглядеть сломанным. Это состояние называют краевым случаем, его стоит обрабатывать как ошибку, например, показывать алерт. С точки зрения проектирования API сервиса IMDB, эту ошибку разработчики должны были вернуть c кодом ошибки клиента [(группа 4хх)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses). Но случается и так, что ответ имеет статус код 200, а тело ответа содержит сообщение об ошибке, как в нашем примере. Такой ответ в любом случае необходимо обработать разработчику фронтенда (web, iOS, android).

Примеры состояний, когда такое происходит:
* API key неверный
* API key просрочен
* количество запросов в день превышено (по документации не больше 100)

tlavr replied:
аа вот оно как! Спасибо! :)

NaR0RoW commented on MovieQuiz/Models/MostPopularMovies.swift:
```diff
@@ -0,0 +1,37 @@
+    let imageURL: URL
+    
+    var resizedImageURL: URL {
+        // создаем строку из адреса
```
Лучше удалять комментарии перед review

NaR0RoW commented on MovieQuiz/Models/QuizQuestion.swift:
```diff
@@ -4,10 +4,11 @@
 struct QuizQuestion {
     // Film name, same as image name in the Assets
-    let image: String
+    let image: Data
```
Когда видишь константу с именем `image`, то ожидаешь увидеть тип `UIImage`. Лучше переименовать название на `imageData`

NaR0RoW commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -142,6 +140,31 @@ final class MovieQuizViewController: UIViewController {
     }
+    
+    private func showLoadingIndicator() {
+        activityIndicator.isHidden = false // говорим, что индикатор загрузки не скрыт
```
Рекомендую посмотреть на свойство [hidesWhenStopped](https://developer.apple.com/documentation/uikit/uiactivityindicatorview/hideswhenstopped) у `activityIndicator`, и использовать его вместо ручного управления свойством `isHidden`.

Когда ты будешь вызывать `startAnimating` – индикатор будет появляться автоматически, соответственно на `stopAnimating` – скрываться, а свойство `isHidden` больше не понадобится, индикатор все сделает за тебя.

Важно: это свойство настраивают всего раз раз: кодом во `viewDidLoad()` или аналогичной галочкой у элемента на сториборде. А строки `activityIndicator.isHidden = false(true)` нужно будет убрать

NaR0RoW commented on MovieQuiz/Services/NetworkClient.swift:
```diff
@@ -0,0 +1,43 @@
+        
+        let task = URLSession.shared.dataTask(with: request) { data, response, error in
+            // Проверяем, пришла ли ошибка
+            if let error = error {
```
`error` можно не создавать, т.к. эта константа не используется

NaR0RoW reviewed (approved):
Отличная работа 🔥

tlavr committed:
fix (comments): Fix comments from code review

-Show activity indicator while loading new movie
-Process movies loading error
-Small code style updates
